File: Automation/001-AutomationFramework/C#/TestAutomation/AppiumCore/Base/BaseScreen.cs
using System;
using OpenQA.Selenium.Appium;

namespace AppiumCore.Base
{
    public class BaseScreen
    {
        public AppiumDriver<AppiumWebElement> AppiumDriver;

        public BaseScreen() => AppiumDriver = DriverFactory.Instance.AppiumDriver;

        public TScreen GetInstance<TScreen>() where TScreen : BaseScreen, new()
        {
            var T = Activator.CreateInstance(typeof(TScreen));
            return (TScreen)T;
        }

        public TScreen As<TScreen>() where TScreen : BaseScreen
        {
            return (TScreen)this;
        }
    }
}

File: Automation/Bosch/TestAutomation/src/main/java/base/BaseScreen.java
package base;

import io.appium.java_client.MobileElement;
import io.appium.java_client.AppiumDriver;
import io.appium.java_client.pagefactory.AppiumFieldDecorator;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.rmi.activation.Activator;

public class BaseScreen {
    public AppiumDriver<MobileElement> appiumDriver;
    public WebDriverWait wait;

    public BaseScreen() {
        appiumDriver = DriverFactory.getInstance().driver;
    }

    public <TScreen extends BaseScreen> TScreen getInstance(Class<TScreen> screenClass) {
        try {
            return screenClass.newInstance();
        } catch (Exception ex) {
            ex.printStackTrace();
            return null;
        }
    }

    public <TScreen extends BaseScreen> TScreen As(Class<TScreen> screenToCast)
    {
        return screenToCast.isInstance(this) ? screenToCast.cast(this) : null;
    }
}
